src/components/elements/TimePicker.js

```diff
@@ -17,7 +18,26 @@ class TimeAndDatePicker extends Component {
 
   constructor () {
     super()
-    this.state = { startDate: null, startTime: null, endDate: null, endTime: null }
+    this.state = {
+      endDate: DateTime.local().toISODate().toString(),
+      endTime: DateTime.local().toISOTime().toString(),
+      startDate: DateTime.local().minus({day: 1}).toISODate().toString(),
+      startTime: DateTime.local().toISOTime().toString()
+    }
+  }
+
+  componentDidMount () {
+    const { filters, dispatch, history } = this.props
+    const { startDate, startTime, endDate, endTime } = this.state
+    const newFilter = Object.assign({}, filters, {startTime: startDate + 'T' + startTime, endTime: endDate + 'T' + endTime})
+    dispatch(filterActions.synchronizeFilters(newFilter, null, null, history))
+  }
+
+  componentDidUpdate (prevProps) {
+    const { filters, dispatch, history } = this.props
+    if (!deepEqual(prevProps.filters, filters)) {
```

Comment:
What cases does `deepEqual` cover that `===` would not?

Reply:
that's a good question, I don't know.

Reply:
I _think_ we can avoid adding a dependency and use the same value equality function [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) here instead of deep equals…

Reply:
Nope, stick with `deepEquals`

<img width="273" alt="screen shot 2018-03-14 at 1 35 47 pm" src="https://user-images.githubusercontent.com/509611/37429504-b969ab98-278c-11e8-8d44-bd6251937d35.png">

Reply:
So I'm not sure if different but identical objects can ever arrive here, but there's nothing explicitly preventing it; depending on the cost of deepEqual, it may make sense to check first whether the objects are the same and then do the deep equality comparison only if they're not actually the same object.

Then again, deepEqual may do that optimization internally anyway.

Reply:
Can we assume that other people's code is optimized until we encounter performance issues? (not because their code is/will be performant, but because premature optimization is the root of all evil.)